modules/local_contacts/modules.php

```diff
@@ -65,6 +65,91 @@ public function process() {
     }
 }
 
+/**
+ * @subpackage local_contacts/handler
+ */
+class Hm_Handler_process_import_contact extends Hm_Handler_Module {
+    public function process() {
+        list($success, $form) = $this->process_form(array('contact_source', 'import_contact'));
+        if ($success && $form['contact_source'] == 'csv') {
+            $file = $this->request->files['contact_csv'];
+            $csv = fopen($file['tmp_name'], 'r');
+            if ($csv) {
+                $contacts = $this->get('contact_store');
+                $header = fgetcsv($csv);
+                $expectedHeader = array('display_name', 'email_address', 'phone_number');
+
+                if ($header !== $expectedHeader) {
+                    fclose($csv);
+                    Hm_Msgs::add('ERRInvalid CSV file, please use a valid header: '.implode(', ', $expectedHeader));
+                    return;
+                }
+
+                $contact_list = $contacts->getAll();
+                $message = '';
+                $update_count = 0;
+                $create_count = 0;
+                $invalid_mail_count = 0;
+                $import_result = [];
+                
+
+                while (($data = fgetcsv($csv)) !== FALSE) {
+                    $single_contact = [
+                        'display_name' => $data[0],
+                        'email_address' => $data[1],
+                        'phone_number' => $data[2] ?? ''
+                    ];
+                    $email = $data[1];
+                    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+                        $single_contact['status'] = 'invalid email';
+                        array_push($import_result, $single_contact);
+                        $invalid_mail_count++;
+                        continue;
+                    }
+
+                    $details = array('source' => 'local', 'display_name' => $data[0], 'email_address' => $email);
+                    if (array_key_exists(2, $data) && $data[2]) {
+                        $details['phone_number'] = $data[2];
+                    }
+
+                    $contactUpdated = false;
+                    foreach ($contact_list as $key => $contact) {
+                        $contact_data = $contact->export();
```

Comment:
You don't need to do contact->export to get just one value, instead use $contact->value('email_address') method. Otherwise, we have too much data passed around that is not used.

Reply:
Updated, thanks!